src/common/thirdparty/build-redis.sh

```diff
@@ -3,9 +3,11 @@
 # Cause the script to exit if a single command fails.
 set -e
 
-if [ ! -f redis-3.2.3/src/redis-server ]; then
-  wget http://download.redis.io/releases/redis-3.2.3.tar.gz
-  tar xvfz redis-3.2.3.tar.gz
-  cd redis-3.2.3
+if [ ! -f redis/src/redis-server ]; then
+  redis_vname="redis-3.2.3"
+  if [ ! -f redis/utils/whatisdoing.sh ]; then
```

Comment:
Haha sorry, I should've explained. It's one of the last files in the tarball, so it's to check the tarball has been extracted fully (and not terminated halfway). The whole PR is because I ran the script multiple times and it didn't really seem idempotent (it'd name the second tarball to `<name>.1` or something) and didn't seem to run properly so I just removed the temporary copy altogether (and stripped the outer directory directly).